piano-hero/mobile/src/main/java/com/novoda/pianohero/SimpleNotesOutputView.java

```diff
@@ -29,54 +26,32 @@ protected void onFinishInflate() {
         messageTextView = (TextView) findViewById(R.id.simple_notes_output_text_message);
     }
 
-    public void display(Sequence sequence) {
-        checkSequenceIsSimpleElseThrow(sequence);
-
-        updateNextNoteToPlay(sequence);
-        updateMessage(sequence);
+    @Override
+    public void showRound(RoundViewModel viewModel) {
+        updateNextNoteToPlay(viewModel);
+        messageTextView.setText(viewModel.getStatusMessage());
     }
 
-    private void checkSequenceIsSimpleElseThrow(Sequence sequence) {
-        for (int i = 0; i < sequence.length(); i++) {
-            Notes notes = sequence.get(i);
-            if (notes.count() > 1) {
-                throw new IllegalArgumentException("Sequence contains chords, that's not simple enough");
-            }
-            for (Note note : notes.notes()) {
-                if (simplePitchNotationFormatter.format(note).endsWith("#")) {
-                    throw new IllegalArgumentException("Sequence contains sharps, that's not simple enough");
-                }
-            }
+    private void updateNextNoteToPlay(RoundViewModel viewModel) {
+        for (String note : viewModel) {
+            nextNoteToPlayTextView.setText(note);
         }
     }
 
-    private void updateNextNoteToPlay(Sequence sequence) {
-        int nextNotesPosition = sequence.position();
-        for (Note note : sequence.get(nextNotesPosition)) {
-            updateNextNoteToPlayTextViewWith(note);
-        }
+    @Override
+    public void showGameComplete() {
+        PianoHeroApplication.popToast("game complete, another!");
     }
 
-    private void updateNextNoteToPlayTextViewWith(Note note) {
-        String simpleNotation = simplePitchNotationFormatter.format(note);
-        nextNoteToPlayTextView.setText(simpleNotation);
+    @Override
+    public void showError() {
+        PianoHeroApplication.popToast("that's not a simple note!");
     }
 
-    private void updateMessage(Sequence sequence) {
-        if (sequence.latestError().count() == 0) {
-            if (sequence.position() > 0) {
-                String message = String.format(Locale.US, "Woo! Keep going! (%d/%d)", sequence.position() + 1, sequence.length());
-                updateMessageTextViewWith(message);
-            } else {
-                messageTextView.setVisibility(INVISIBLE);
-            }
-        } else {
-            updateMessageTextViewWith("Ruhroh, try again!");
-        }
+    @Override
+    public void showKeyboard() {
```

Comment:
GameMvp.View has a method called `showKeyboard` which is dependent on platform/view. Most of the Views will not support this